web/modules/custom/asset_release/src/Controller/AssetReleaseController.php

```diff
@@ -73,10 +99,7 @@ public function add(RdfInterface $rdf_entity) {
    *   The access result object.
    */
   public function createAssetReleaseAccess(RdfInterface $rdf_entity) {
-    $user = $this->currentUser();
-    $membership = Og::getMembership($user, $rdf_entity);
-    // @todo: Remove check for empty after ISAICP-2369 is in.
-    return (!empty($membership) && $membership->hasPermission('create asset_release rdf_entity')) ? AccessResult::allowed() : AccessResult::forbidden();
+    return $this->ogAccess->userAccessEntity('create', $this->createNewAssetRelease($rdf_entity), $this->currentUser());
```

Comment:
> Note that you cannot use OgAccess::userAccess() for this. This is intended to check a user's access to operations on groups. It doesn't work with group content.

This is exactly what we are trying to do here: we are checking if inside the {rdf_entity} parameter aka solution, which is the group, we have the permission to create a release, which is the group content entity.
We don't yet have any group content, so we cannot say that we are working with group content.
This is also how it is done in og for Drupal 7, for example to render node create links.

I agree that (at least for now, depending on the og_context future implementation) we cannot use `hook_entity_create_access` because of missing context.
Still, I can see that the old `node_access()` (that naturally became an `entity_access`) has been split in an `entity_create_access`, which takes a bundle as parameter, and an `entity_access`, which takes the entity (this is a simplification of the actual implementation but we can get the picture).
This clearly defines for me the difference of permission checking during a creation and any other ~~C~~RUD operation on entities.

So that's why I still think we should check the permission on the group and not the permission on the group content.